packages/nc-gui/composables/useMultiSelect/index.ts

```diff
@@ -0,0 +1,268 @@
+    if (
+      !isNaN(selectedRows.startRow) &&
+      !isNaN(selectedRows.startCol) &&
+      !isNaN(selectedRows.endRow) &&
+      !isNaN(selectedRows.endCol)
+    ) {
+      // In case the user press tabs or arrows keys
+      selected.row = selectedRows.startRow
+      selected.col = selectedRows.startCol
+    }
+
+    if (selected.row === null || selected.col === null) return
+
+    /** on tab key press navigate through cells */
+    switch (e.key) {
+      case 'Tab':
+        e.preventDefault()
+        clearRangeRows()
+
+        if (e.shiftKey) {
+          if (selected.col > 0) {
+            selected.col--
+          } else if (selected.row > 0) {
+            selected.row--
+            selected.col = unref(columnLength) - 1
+          }
+        } else {
+          if (selected.col < unref(columnLength) - 1) {
+            selected.col++
+          } else if (selected.row < unref(data).length - 1) {
+            selected.row++
+            selected.col = 0
+          }
+        }
+        break
+      /** on enter key press make cell editable */
+      case 'Enter':
+        e.preventDefault()
+        clearRangeRows()
+        makeEditable(unref(data)[selected.row], unref(fields)[selected.col])
+        break
+      /** on delete key press clear cell */
+      case 'Delete':
+        if (!unref(editEnabled)) {
+          e.preventDefault()
+          clearRangeRows()
+          await clearCell(selected as { row: number; col: number })
+        }
+        break
+      /** on arrow key press navigate through cells */
+      case 'ArrowRight':
+        e.preventDefault()
+        clearRangeRows()
+        if (selected.col < unref(columnLength) - 1) selected.col++
+        break
+      case 'ArrowLeft':
+        clearRangeRows()
+        e.preventDefault()
+        clearRangeRows()
+        if (selected.col > 0) selected.col--
+        break
+      case 'ArrowUp':
+        clearRangeRows()
+        e.preventDefault()
+        clearRangeRows()
+        if (selected.row > 0) selected.row--
+        break
+      case 'ArrowDown':
+        clearRangeRows()
+        e.preventDefault()
+        clearRangeRows()
+        if (selected.row < unref(data).length - 1) selected.row++
+        break
+      default:
+        {
+          const rowObj = unref(data)[selected.row]
+
+          const columnObj = unref(fields)[selected.col]
+
+          let cptext = '' // variable for save the text to be copy
+
+          if (!isNaN(rangeRows.minRow) && !isNaN(rangeRows.maxRow) && !isNaN(rangeRows.minCol) && !isNaN(rangeRows.maxCol)) {
+            const cprows = unref(data).slice(rangeRows.minRow, rangeRows.maxRow + 1) // slice the selected rows for copy
+
+            const cpcols = unref(fields).slice(rangeRows.minCol, rangeRows.maxCol + 1) // slice the selected cols for copy
+
+            cprows.forEach((row) => {
+              cpcols.forEach((col) => {
+                cptext = `${cptext} ${row.row[col.title]} \t`
```

Comment:
- LTAR and Lookup column value will copy as  `[object Object]` 
- Attachment will be JSON string or `[object Object]`

Reply:
Should these be filtered out?

Reply:
Either we can filter or we need to extract the primary value and concatenate. Or even we can convert to JSON string to begin with and later we can do the primary value extracton.

Reply:
@elvus Let me fix the copy issue for the mentioned cells and then we can get this merged :)